Assets/Scripts/Tank/TankService.cs

```diff
@@ -1,37 +1,97 @@
-﻿using UnityEngine;
+﻿using System.Collections.Generic;
+using UnityEngine;
 public class TankService : MonoSingleton<TankService>
 {
-    public TankView tankView;
     private Camera mainCamera;
-    private TankModel tankModel;
-    private TankController tankController;
-    protected Joystick stick;
+
+    [SerializeField]
+    private List<EnemyController> enemyList;
+    [SerializeField]
+    private List<Transform> spawnPositions;
+    [SerializeField]
+    private SpawnerObject spawnerObject;
+    public List<TankScriptableObject> enemiesToGenerate;
+    public Transform enemyParent;
+    public EnemyController redTank;
+    public EnemyController blueTank;
+    public EnemyController greenTank;
+
+    [SerializeField]
+    private TankController playerTank;
+
+    public Transform PlayerTransform{
+        get{
+            Transform playerTransform = playerTank.transform;
+            return playerTransform;
+        }
+    }
+
     private void Start() {
-        GetJoystick();
-        CreateTank();
-        AssignCamera();
+        populateSpawnPositions();
+        CreatePlayer();
+        EnemyController createdEnemy;
+        createdEnemy = CreateEnemy(enemiesToGenerate[0]);
+        enemyList.Add(createdEnemy);
+        createdEnemy = CreateEnemy(enemiesToGenerate[1]);
+        enemyList.Add(createdEnemy);
+        createdEnemy = CreateEnemy(enemiesToGenerate[2]);
+        enemyList.Add(createdEnemy);
+    }
+
+    // Player Creation...
+    private void CreatePlayer(){
+        Vector3 randomSpawnPosition = GetRandomSpawnPos();
+        playerTank.transform.position = randomSpawnPosition;
+        Joystick joystick = GetJoystick();
+        AssignJoystick(playerTank, joystick);
+        AssignCamera(playerTank);
     }
 
-    private void GetJoystick(){
+    private Joystick GetJoystick(){
         InputService joystickInstance = InputService.Instance;
-        stick = joystickInstance.Joystick;
+        Joystick tempJoystick;
+        tempJoystick = joystickInstance.Joystick;
+        return tempJoystick;
     }
 
-    #region Tank Creator
-        private void CreateTank(){
-            tankController = CreateTankController(stick);
-        }
-        private TankController CreateTankController(Joystick stick){
-            tankModel = new TankModel(10f, 30f, 100);
-            tankController = new TankController(tankModel, tankView, stick);
-            return tankController;
-        }
-    #endregion
+    private void AssignJoystick(TankController tankController,  Joystick joystick){
+        tankController.Joystick = joystick;
+    }
 
-    private void AssignCamera(){
+    private void AssignCamera(TankController player){
         mainCamera = Camera.main;
         GameObject playerObj;
-        playerObj = tankController.TankView.gameObject;
+        playerObj = player.gameObject;
         mainCamera.gameObject.GetComponent<PlayerTracker>().PlayerTransform = playerObj.GetComponent<Transform>();
```

Comment:
You already have a variable for playerTransform. Even playerObj is not needed

Reply:
Ya removed!